e2e/internal/e2e/keyexec.go

```diff
@@ -0,0 +1,204 @@
+func RemoveDefaultPublicKey(t *testing.T) {
+	LoadEnv(t, &testenv)
+
+	argv := []string{"key", "remove", "F69C21F759C8EA06FD32CCF4536523CE1E109AF3"}
+	execKey := exec.Command(testenv.CmdPath, argv...)
+
+	out, err := execKey.CombinedOutput()
+	if err != nil {
+		t.Log(string(out))
+		t.Fatalf("Unable to pull key: %v", err)
+	}
+}
+
+// BackupSecretKeyring will take your secret keyring file, and back it up. This gets ran before the private
+// key testing.
+func BackupSecretKeyring(t *testing.T) {
+	err := os.Rename(filepath.Join(HomeDir(), ".singularity/sypgp/pgp-secret"), backupSypgp)
+	if err != nil {
+		t.Fatalf("Unable to rename secret keyring: %v", err)
+	}
+}
+
+// RecoverSecretKeyring will recover your secret keyring, this gets ran after the private key test are complete.
+func RecoverSecretKeyring(t *testing.T) {
+	if err := os.Remove(filepath.Join(HomeDir(), ".singularity/sypgp/pgp-secret")); err != nil {
+		t.Fatalf("Unable to remove secret keyring: %v", err)
+	}
+	if err := os.Rename(backupSypgp, filepath.Join(HomeDir(), ".singularity/sypgp/pgp-secret")); err != nil {
+		t.Fatalf("Unable to rename secret keyring: %v", err)
+	}
+}
+
+// RemoveSecretKeyring will delete your secret keyring :O
+func RemoveSecretKeyring(t *testing.T) {
+	err := os.Remove(filepath.Join(HomeDir(), ".singularity/sypgp/pgp-secret"))
+	if err != nil {
+		t.Fatalf("Unable to remove secret keyring: %v", err)
+	}
+}
+
+// ImportKey will import a key from kpath.
+func ImportKey(t *testing.T, kpath string) ([]byte, error) {
+	LoadEnv(t, &testenv)
+
+	argv := []string{"key", "import", kpath}
+	execKey := exec.Command(testenv.CmdPath, argv...)
+
+	return execKey.CombinedOutput()
+}
+
+// ImportTestKey will take a private key file (kpath) and import it.
+func ImportPrivateKey(t *testing.T, kpath string) ([]byte, error) {
+	s := getImportScript(kpath)
+
+	importScript, err := ioutil.TempFile("", "")
+	if err != nil {
+		t.Fatalf("Unable to create script: %v", err)
+	}
+	defer importScript.Close()
+
+	err = ioutil.WriteFile(importScript.Name(), []byte(s), 0644)
+	if err != nil {
+		t.Fatalf("Unable to write tmp file: %v", err)
+	}
+
+	argv := []string{importScript.Name()}
+	execImport := exec.Command("/usr/bin/expect", argv...)
+
+	return execImport.CombinedOutput()
+}
+
+// ExportPrivateKey will import a private key from kpath.
+func ExportPrivateKey(t *testing.T, kpath string) ([]byte, error) {
+	s := getExportScript(0, kpath, "", "e2etests")
+
+	exportScript, err := ioutil.TempFile("", "")
+	if err != nil {
+		t.Fatalf("Unable to create script: %v", err)
+	}
+	defer exportScript.Close()
+
+	_, err = exportScript.WriteString(s)
+	if err != nil {
+		t.Fatalf("Unable to write to file: %v", err)
+	}
+	err = exportScript.Close()
+	if err != nil {
+		t.Fatalf("Unable to clise file: %v", err)
+	}
+
+	argv := []string{exportScript.Name()}
+	execImport := exec.Command("/usr/bin/expect", argv...)
+
+	return execImport.CombinedOutput()
```

Comment:
you should check the exit code of the command, no?

Reply:
This is checked where the function is called, in some test cases, this is supposed to fail.